### PR TITLE
BYOI for OpenPhish database

### PR DESCRIPTION
**Pull request by gshriki**
OpenPhish uses proprietary Artificial Intelligence algorithms to automatically identify zero-day phishing sites and provide comprehensive, actionable, real-time threat intelligence.

This BYOI includes the `url` command to check if a URL is in the OpenPhish database.

### CODE REVIEW

File: Integrations/integration-OpenPhish.yml [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+        }
+        var res = {'url':args.url, 'match': true};
+        return {Type: entryTypes.note, Contents: res, ContentsFormat: formats.json, HumanReadable: md, EntryContext: ec};
+    }
+    switch (command) {
+        case 'test-module':
+            if (!isNumber(params.fetchIntervalHours)) {
+                throw 'OpenPhish error: Please provide a numeric value (and bigger than 0) for Database refresh interval (hours)';
+            }
+            doURL('http://google.com');
```
Comment (aviadl) [inline Integrations/integration-OpenPhish.yml]: why do you need this in the test? 
Test needs to validate that the you can download the list now ?

File: Integrations/integration-OpenPhish.yml [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+    var reload = function() {
+        var list = getList();
+
+        var data={'list':list,'timestamp':new Date().getTime()};
+        setIntegrationContext(data);
+        return data;
+    };
+
+    var reloadNeeded = function(data) {
```
Comment (aviadl) [inline Integrations/integration-OpenPhish.yml]: I would change to `isReloadNeeded`

File: Integrations/integration-OpenPhish.yml [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+        var res = {'url':args.url, 'match': true};
+        return {Type: entryTypes.note, Contents: res, ContentsFormat: formats.json, HumanReadable: md, EntryContext: ec};
+    }
+    switch (command) {
+        case 'test-module':
+            if (!isNumber(params.fetchIntervalHours)) {
+                throw 'OpenPhish error: Please provide a numeric value (and bigger than 0) for Database refresh interval (hours)';
+            }
+            doURL('http://google.com');
+            data=reload();
```
Comment (aviadl) [inline Integrations/integration-OpenPhish.yml]: I would not call `reload`, but rather call `getList`

File: Integrations/integration-OpenPhish.yml [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+        var url=pUrl;
+        if (url[url.length - 1] == '/') {
+            url = url.substr(0,url.length - 1);
+        }
+        return url;
+    }
+    var doURL = function(pUrl) {
+        var data = getIntegrationContext();
+        if (reloadNeeded(data)) {
+            data = {timestamp:new Date().getTime(), list: getList()};
```
Comment (aviadl) [inline Integrations/integration-OpenPhish.yml]: call `reload` here ?

**Comment by aviadl** [comment]
Dont forget to add a test playbook

**Comment by gshriki** [comment]
Added test playbook

**Pull request merged**